packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -1713,3 +1733,312 @@ describe( 'Post types', () => {
 		}
 	);
 } );
+
+describe( 'Rich link previews', () => {
+	const selectedLink = {
+		id: '1',
+		title: 'Wordpress.org', // customize this for differentiation in assertions
+		url: 'https://www.wordpress.org',
+		type: 'URL',
+	};
+
+	beforeAll( () => {
+		/**
+		 * These tests require that we exercise the `fetchRemoteUrlData` function.
+		 * We are therefore overwriting the mock "placeholder" with a true jest mock
+		 * which will cause the code under test to execute the code which fetches
+		 * rich previews.
+		 */
+		mockFetchRemoteUrlData = jest.fn();
+	} );
+
+	it( 'should not fetch or display rich previews by default', async () => {
+		mockFetchRemoteUrlData.mockImplementation( () =>
+			Promise.resolve( {
+				title:
+					'Blog Tool, Publishing Platform, and CMS \u2014 WordPress.org',
+				icon: 'https://s.w.org/favicon.ico?2',
+				description:
+					'Open source software which you can use to easily create a beautiful website, blog, or app.',
+				image: 'https://s.w.org/images/home/screen-themes.png?3',
+			} )
+		);
+
+		act( () => {
+			render( <LinkControl value={ selectedLink } />, container );
+		} );
+
+		// mockFetchRemoteUrlData resolves on next "tick" of event loop
+		await act( async () => {
+			await eventLoopTick();
+		} );
```

Comment:
IMO, _when_ the link previews show is an implementation detail, we only care about whether they show or not. Instead, we can use the `findBy*` queries from `@testing-library/react` to query the resolved UI.

Reply:
I had no idea we had access to RTL. I would _much_ prefer to use those utils. That said, the pattern in these tests currently is to use the `eventLoopTick()`. Could we do this in a followup to avoid making this PR any larger?

Reply:
Oh I didn't know that, sure! Absolutely.